packages/shared/lib/services/sync/config/deploy.service.ts

```diff
@@ -204,6 +217,162 @@ export async function deploy({
     }
 }
 
+export async function upgradePreBuilt({
+    environment,
+    account,
+    flowConfig,
+    logContextGetter
+}: {
+    environment: Environment;
+    account: Account;
+    flowConfig: IncomingFlowConfigUpgrade;
+    logContextGetter: LogContextGetter;
+}): Promise<Result<boolean | null>> {
+    const logCtx = await logContextGetter.create(
+        { operation: { type: 'deploy', action: 'prebuilt' }, message: 'Upgrading pre-built flow' },
+        { account, environment }
+    );
+
+    const syncConfig = await getNangoConfigIdAndLocationFromId(Number(flowConfig.id));
+
+    if (!syncConfig) {
+        const error = new NangoError('unknown_sync_config', { syncConfigId: flowConfig.id });
+
+        return Err(error);
+    }
+
+    const config = await configService.getById(syncConfig.nango_config_id);
+
+    if (!config) {
+        const error = new NangoError('unknown_provider_config');
+
+        return Err(error);
+    }
+
+    const { syncName: name, type, model_schema: model_schema_string, is_public, upgrade_version: version } = flowConfig;
+    const { unique_key: provider_config_key, provider } = config;
+
+    const file_location = (await remoteFileService.copy(
+        `${provider}/dist`,
+        `${name}-${provider}.js`,
+        `${env}/account/${account.id}/environment/${environment.id}/config/${syncConfig.nango_config_id}/${name}-v${version}.js`,
+        environment.id,
+        `${name}-${provider_config_key}.js`
+    )) as string;
+
+    if (!file_location) {
+        await logCtx.error('There was an error uploading the template', { isPublic: is_public, syncName: name, version });
+        await logCtx.failed();
+
+        throw new NangoError('file_upload_error');
+    }
+
+    await remoteFileService.copy(
+        provider,
+        `${type}s/${name}.ts`,
+        `${env}/account/${account.id}/environment/${environment.id}/config/${syncConfig.nango_config_id}/${name}.ts`,
+        environment.id,
+        `${name}.ts`
+    );
+
+    const created_at = new Date();
+
+    const model_schema = typeof model_schema_string === 'string' ? JSON.parse(model_schema_string) : model_schema_string;
+
+    const flowData: SyncConfig = {
+        created_at,
+        sync_name: name,
+        sync_type: flowConfig.sync_type,
+        runs: flowConfig.runs,
+        models: flowConfig.models,
+        metadata: flowConfig.metadata || {},
+        nango_config_id: syncConfig.nango_config_id,
+        file_location,
+        version,
+        active: true,
+        model_schema: JSON.stringify(model_schema) as unknown as SyncModelSchema[],
```

Comment:
Is it useful to parse it above to then stringify it here?

Reply:
Yeah, necessary to insert it correctly into the jsonb column